js/components/MilkdropWindow/index.js

```diff
@@ -0,0 +1,103 @@
+import React from "react";
+import screenfull from "screenfull";
+import butterchurn from "butterchurn";
+import reactionDiffusion2 from "butterchurn-presets/presets/converted/Geiss - Reaction Diffusion 2";
+
+class MilkdropWindow extends React.Component {
+  constructor() {
+    super();
+    this.state = {};
+  }
+  componentDidMount() {
+    const scale = window.devicePixelRatio;
+    const analyserNode = this.props.analyser;
+    this.visualizer = butterchurn.createVisualizer(
+      analyserNode.context,
+      this._canvasNode,
+      {
+        width: this._canvasNode.width * scale,
+        height: this._canvasNode.height * scale
+      }
+    );
+    this.visualizer.connectAudio(analyserNode);
+    this.visualizer.loadPreset(reactionDiffusion2);
+    this._renderViz();
+
+    this._canvasNode.addEventListener("dblclick", () => this._handleRequestFullsceen());
+
+    require.ensure(['butterchurn-presets'], require => {
+      const butterchurnPresets = require('butterchurn-presets');
+      const presets = butterchurnPresets.getPresets();
+      const presetKeys = Object.keys(presets);
+      this.cycleInterval = setInterval(() => {
+        const preset =
+          presets[presetKeys[Math.floor(presetKeys.length * Math.random())]];
+        this.visualizer.loadPreset(preset);
+      }, 15000);
+    });
+  }
+  componentWillUnmount() {
+    this._pauseViz();
+    this._stopCycling();
+  }
+  componentDidUpdate() {
+    if (this.props.width !== this.state.width || this.props.height !== this.state.height) {
```

Comment:
Can't remember the right way to do this hah, I only want to update the renderer size when the width/height has actually changed